app/controllers/proctored_exams_controller.rb

```diff
@@ -32,20 +32,55 @@ def start_quiz
       proctor_code: params[:proctor_code],
     }.to_query
 
-    quiz = HTTParty.get(
-      "#{plugin.settings[:adhesion_url]}/api/proctored_exams?#{query}",
+    exam_request = HTTParty.get(
+      "#{plugin.settings[:adhesion_proctor_url]}/api/proctored_exams?#{query}",
       headers: headers,
       # verify: false,
-    )
-    if quiz.parsed_response["error"].present?
-      flash[:error] = quiz.parsed_response["error"]
+    ).parsed_response["exam_request"]
+
+    if exam_request.nil?
+      flash[:error] = "You do not have an exam that is ready to start."
+      redirect_to proctored_exams_path
+      return
+    end
+
+    matched_code = false
+    proctor_id = nil
+    proctor_name = nil
+    code = params[:proctor_code]
+    Account.find(exam_request["testing_center_id"]).users.find_each do |user|
```

Comment:
This would make this less hard on the database:
```ruby
user_ids = Account.find(exam_request["testing_center_id"]).users.pluck(:id)
CustomData.where(user_id: user_ids, namespace: "edu.au.exam").find_each do |cd|
  if cd.data["d"]["exam"]["proctor_code"] == code
    matched_code = true
    proctor_id = user.id
    proctor_name = user.name
    break
  end
end
```
Can you try that?

Reply:
I can. How does this make it easier on the database though, im curious?

Reply:
It does only two database hits (probably). One to get all the user_ids. Then another to get all the custom data (up to 1000 records at a time, this is where the probably comes into play), and then loops over that. Whereas currently it does one to get all the users. Then one fore every user to find a custom data for them.

Reply:
Ok that makes sense.

Reply:
```ruby
users = Account.find(exam_request["testing_center_id"]).users
CustomData.where(user: users, namespace: "edu.au.exam").find_each do |cd|
  user = users.detect { |u| u.id == cd.user_id }
  if cd.data["d"]["exam"]["proctor_code"] == code
    matched_code = true
    proctor_id = user.id
    proctor_name = user.name
    break
  end
end
```
How about that?